rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/10_histogram.yml

```diff
@@ -115,31 +106,394 @@ setup:
         body: { "aggs" : { "histo" : { "histogram" : { "field" : "number", "interval" : 50, "format" : "Value is ##0.0" } } } }
 
   - match: { hits.total: 4 }
+  - length: { aggregations.histo.buckets: 4 }
+
+  - match: { aggregations.histo.buckets.0.key: 0 }
+  - match: { aggregations.histo.buckets.0.key_as_string: "Value is 0.0" }
+  - match: { aggregations.histo.buckets.0.doc_count: 1 }
+
+  - match: { aggregations.histo.buckets.1.key: 50 }
+  - match: { aggregations.histo.buckets.1.key_as_string: "Value is 50.0" }
+  - match: { aggregations.histo.buckets.1.doc_count: 1 }
+
+  - match: { aggregations.histo.buckets.2.key: 100 }
+  - match: { aggregations.histo.buckets.2.key_as_string: "Value is 100.0" }
+  - match: { aggregations.histo.buckets.2.doc_count: 1 }
+
+  - match: { aggregations.histo.buckets.3.key: 150 }
+  - match: { aggregations.histo.buckets.3.key_as_string: "Value is 150.0" }
+  - match: { aggregations.histo.buckets.3.doc_count: 1 }
+
+---
+"Show empty intervals":
+  - do:
+      index:
+        index: test_1
+        id: 1
+        body: { "number" : 1 }
+
+  - do:
+      index:
+        index: test_1
+        id: 2
+        body: { "number" : 51 }
+
+  - do:
+      index:
+        index: test_1
+        id: 3
+        body: { "number" : 61 }
+
+  - do:
+      index:
+        index: test_1
+        id: 4
+        body: { "number" : 71 }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        body: { "aggs" : { "histo" : { "histogram" : { "field" : "number", "interval" : 20 } } } }
```

Comment:
I think it's easier to read when these are on individual lines. I tend to write them in yaml instead of json too just because we're in yaml files. But I think that is less important than the line feeds.